STM32/LightController/Core/Src/LightController.c

```diff
@@ -76,9 +92,48 @@ static void LightControllerStatus()
     writeUSB(buf, len);
 }
 
+static void updateLEDCtrl(int channel, unsigned int red, unsigned int green, unsigned int blue, int whiteOn)
+{
+    if (whiteOn)
+    {
+        rgbwControl[channel*NO_COLORS] 	   = 0;
+        rgbwControl[channel*NO_COLORS + 1] = 0;
+        rgbwControl[channel*NO_COLORS + 2] = 0;
+        rgbwControl[channel*NO_COLORS + 3] = MAX_PWM; 
+        return;
+    }
+
+    rgbwControl[channel*NO_COLORS] 	   = red;
+    rgbwControl[channel*NO_COLORS + 1] = green;
+    rgbwControl[channel*NO_COLORS + 2] = blue;
+    rgbwControl[channel*NO_COLORS + 3] = 0; 
+}
+
 bool isInputValid(const char *input, int *channel, unsigned int *rgb)
 {
-   
+    // If PARTY command is entered start the colour test
+    // If any other input is entered stop it again
+    if (strcmp(input, "PARTY") == 0) {
+        teststate = true;
+        return true;
+    }
+    else{
+        // Reset the colours when disabling teststate
+        if(teststate=true){
+        int red = 0;
+        int green = 0;
+        int blue = 0;
+        int whiteon = false;
+        for (int channel = 0;channel < 3;channel++){
+            updateLEDCtrl(channel,red,green,blue,whiteon);
+        };
```

Comment:
Reading how the rest of the function is written and used, I don't think it was intended to have side-effects, and this introduces one. I would remove this entire if statement.

Reply:
This is the only thing I've read by the end of the review that might cause the unit test failure by the way.

Reply:
Done